File: sample/java/BubbleSort.java
// Assisted by watsonx Code Assistant 
public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = {5, 2, 9, 1, 7};
        int n = arr.length;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }

        System.out.println("Sorted array: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}

File: sample/perl/bubblesort.pl
# Assisted by watsonx Code Assistant 
# watsonx Code Assistant did not check whether this code suggestion might be similar to third party code.
 

sub bubblesort {
    my @array = @_;
    my $length = @array;
    for (my $i = 0; $i < $length; $i++) {
        for (my $j = 0; $j < $length - $i - 1; $j++) {
            if ($array[$j] > $array[$j + 1]) {
                my $temp = $array[$j];
                $array[$j] = $array[$j + 1];
                $array[$j + 1] = $temp;
            }
        }
    }
    return @array;
}

print "Enter the array elements: ";
my @array = <STDIN>;
chomp(@array);

@array = bubblesort(@array);

print "Sorted array: ";
print "@array\n";